challenge_4/python/OkraArmband/src/challenge_4.py

```diff
@@ -0,0 +1,49 @@
+class BinaryTree():
+    def __init__(self, root):
+        self.root = root
+        self.left = None
+        self.right = None
+
+    def invertTree(self):
+        """
+        Swap left and right children of input BinaryTree for all subsequent children as well (recursively).
+        """
+        temp = self.left
+        self.left = self.right
+        self.right = temp
+
+
+        # Continue the recursion as long as we haven't reached the end of a branch
+        if self.left is not None:
+            self.left.invertTree()
+        if self.right is not None:
+            self.right.invertTree()
+
+    @classmethod
+    def from_list(cls,data):
```

Comment:
It's clear that you've set a precondition for the user and while it's unlikely that someone like yourself would break that precondition, it is possible for someone to pass in `[2, 3]` in which case `root = cls(data)` would make `root` an array.